loopdb/sql_store.go

```diff
@@ -630,6 +752,170 @@ func (s *BaseDB) convertLoopInRow(row sqlc.GetLoopInSwapsRow,
 	return loopIn, nil
 }
 
+// convertBatchRow converts a batch row from db to a sweepbatcher.Batch struct.
+func (s *BaseDB) convertBatchRow(row sqlc.SweepBatch) (Batch, error) {
+	batch := Batch{
+		ID:    row.ID,
+		State: row.State,
+	}
+
+	if row.BatchTxID.Valid {
+		err := chainhash.Decode(&batch.BatchTxid, row.BatchTxID.String)
+		if err != nil {
+			return batch, err
+		}
+	}
+
+	batch.BatchPkScript = row.BatchPkScript
+
+	if row.LastRbfHeight.Valid {
+		batch.LastRbfHeight = row.LastRbfHeight.Int32
+	}
+
+	if row.LastRbfSatPerKw.Valid {
+		batch.LastRbfSatPerKw = row.LastRbfSatPerKw.Int32
+	}
+
+	batch.MaxTimeoutDistance = row.MaxTimeoutDistance
+
+	return batch, nil
+}
+
+// BatchToUpsertArgs converts a Batch struct to the arguments needed to insert
+// it into the database.
+func (s *BaseDB) BatchToInsertArgs(
+	batch Batch) sqlc.InsertBatchParams {
+
+	args := sqlc.InsertBatchParams{
+		State: batch.State,
+		BatchTxID: sql.NullString{
+			Valid:  true,
+			String: batch.BatchTxid.String(),
+		},
+		BatchPkScript: batch.BatchPkScript,
+		LastRbfHeight: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfHeight,
+		},
+		LastRbfSatPerKw: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfSatPerKw,
+		},
+		MaxTimeoutDistance: batch.MaxTimeoutDistance,
+	}
+
+	return args
+}
+
+// BatchToUpsertArgs converts a Batch struct to the arguments needed to insert
+// it into the database.
+func (s *BaseDB) BatchToUpdateArgs(
```

Comment:
This function can be private and doesn't need `BaseDB` receiver either since it's not using any members.